server/container/guice/guice-common/src/main/java/org/apache/james/utils/FileConfigurationProvider.java

```diff
@@ -58,8 +58,11 @@ public HierarchicalConfiguration getConfiguration(String component) throws Confi
         Preconditions.checkNotNull(component);
         List<String> configPathParts = Splitter.on(".").splitToList(component);
         Preconditions.checkArgument(!configPathParts.isEmpty());
-        HierarchicalConfiguration config = getConfig(retrieveConfigInputStream(configPathParts.get(0)));
-        return selectHierarchicalConfigPart(config, Iterables.skip(configPathParts, 1));
+        return getConfiguration(retrieveConfigInputStream(configPathParts.get(0)), Iterables.skip(configPathParts, 1));
+    }
+
+    public HierarchicalConfiguration getConfiguration(InputStream inputStream, Iterable<String> configsPathParts) throws ConfigurationException {
```

Comment:
second answer : 
1/ you are not testing what the application will use
2/ @VisibleForTesting is more appropriate
3/ the prototype is just plain wrong : what inputstream is supposed to contain ? Iterable<String> configsPathParts ? it's a hack, no a method you expose, even for testing purpose. If you need it, make it usable first